geniushubclient/zone.py

```diff
@@ -33,15 +34,128 @@ def alphanum_key(k):
     return sorted(dict_list, key=alphanum_key)
 
 
+@dataclass
+class CurrentState:
+    """The current state of the zone.
+
+        Intended to be properties that can change in the lifecycle of the zone.
+    """
+
+    is_active: bool = False
+    is_requesting_heat: bool = False
+    mode: int = 0
+    mode_name: str = ""
+    temperature: float = 0.0
+    set_point: float = 0.0
+    is_occupied: bool = False
+
+
+@dataclass
+class Properties:
+    """The properties of the zone.
+
+        Intended to be the properties that are fixed for the lifecycle of the zone
+    """
+
+    type: int = 0
+    type_name: str = ""
+    has_room_sensor: bool = False
```

Comment:
Ah, missed this one in my update. I must admit, I assumed that has_pir basically meant whether the room has a sensor, so I just wanted to make that more obvious. Does it have other connotations then?

Reply:
There are no connotations to `has_pir`.

It's the leading _ which is the main thing (changing the name is only a 2' issue) - I was using the leading _ to imply a name private to the library - use it if you will, but it's not a public name, and no promises made it will hang around (since it is not in the public API yet, and so may change).

My preference is to keep the name for now only for the usual reasons (HA code currently uses it), and the leading _ for the reasons above.